### PR TITLE
[ARCH] 01 - Sass lib -> src & Yarn workspaces

### LINKED ISSUE #602: Hoist node_modules to root folder
Using either `Bolt` or `Yarn` we need a way to hoist our `node_modules/` folder to the root directory, this will radically improve performance and disk usage.

- [x] Change building from `lib/` to `src/`
- [x] Remove `lerna`

### PR DESCRIPTION
**Pull request by adamzerella**
Fix #602 

**CHANGES**
- `packages` -> `components`
- delete `.cfignore` - not used anymore
- components now read from `src/` over `lib/`
- Setup Yarn workspaces to hoist `node_modules` into shared root folder
- CircleCI caching of root `node_modules/`
- `mocha` over `jest` for animate unit tests
- Upgraded all packages to latest e.g `webpack 3` -> `webpack 4` - This includes moving to `@babel/7` which brings some really good optimizations

**ISSUES**
- We are losing `sass-verioning` functionality with this change. Reading our sass files from `src/` results in versioning to fail as we are now checking files that haven't had their dependencies set. We should look at bringing this or an alternative of this back sometime in the future.

**NOTES**
- This branch fails to build in Windows due to pathing complications from `pancake`. We should look at resolving this but our Windows support in general is something else to discuss. 

- This PR will require publishing new changes to the following packages:
```
- cta-link
- direction-links
- header
- main-nav
- side-nav
- skip-link
- accordion
- body
```
due to the minor changes of updating prefixer causing  `text-decoration-skip: ink;` -> `text-decoration-skip-ink: auto;`

### CODE REVIEW

**Comment by adamzerella** [comment]
@alex-page Ready for another pass, only remaining thing is puppeteer running in circle and making helpers async

File: scripts/a11y.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ const DisplayResults = results => {
 			console.log( `${ issue.context }\n ${ issue.message }\n` );
 		})
 
-		process.exit( 1 );
+		// process.exit( 1 );
```
Comment (alex-page) [inline scripts/a11y.js]: Harden up mate 😂

File: components/accordion/src/sass/_module.scss [hunk 2]
```diff
@@ -55,7 +55,7 @@
 	display: block;
 	color: $AU-color-foreground-action;
 	text-decoration: underline;
-	text-decoration-skip: ink;
+	text-decoration-skip: auto;
```
[the remark this answers is not in the document]
Comment (adamzerella) [reply components/accordion/src/sass/_module.scss]: The warning was generated for `text-decoration-skip` advising to change this from `ink` to `auto` ? I'm not that good with these prefixes we can add it back and talk about the warning later

[the remark this answers is not in the document]
Comment (adamzerella) [reply components/accordion/src/sass/_module.scss]: ...Weird, now the warning isn't generated when building the accordion?

[the remark this answers is not in the document]
Comment (adamzerella) [reply components/accordion/src/sass/_module.scss]: My bad, resolved.

File: components/accordion/package.json [hunk 3]
```diff
@@ -16,20 +16,15 @@
 	],
 	"scripts": {
 		"postinstall": "pancake",
-
```
[the remark this answers is not in the document]
Comment (adamzerella) [reply components/accordion/package.json]: Never removed it 😢 git is being weird

File: .templates/new-module/package.json [hunk 4]
```diff
@@ -45,18 +45,18 @@
...
+				"path": "src/js/module.js"
 			},
 			"react": {
-				"path": "lib/js/react.js"
+				"path": "src/js/react.js"
 			}
 		}
 	},
-	"main": "lib/js/react.es5.js",
+	"main": "src/js/react.js",
```
[the remark this answers is not in the document]
Comment (alex-page) [reply .templates/new-module/package.json]: Yes, however the imports will then default to `lib` so we will need to specify the path.

File: components/accordion/src/sass/_dependencies.scss [hunk 5]
```diff
@@ -0,0 +1,10 @@
+//sass-versioning
+@import '../../../../node_modules/sass-versioning/dist/_index.scss';
+//dependencies
+@import '../../../core/src/sass/_module.scss';
+
+//this module
+@import '../../src/sass/_module.scss';
+
+//version check with sass-versioning
+// @include versioning-check();
```
[the remark this answers is not in the document]
Comment (adamzerella) [reply components/accordion/src/sass/_dependencies.scss]: 😕 Good point, honestly I have no idea this file is identical to the original `_dependencies.scss` with the exception of the commented out versioning-check...

**Pull request merged**